envutil/envutil.go

```diff
@@ -15,6 +15,32 @@ func RevokableSetenv(envKey, envValue string) (func() error, error) {
 	return revokeFn, os.Setenv(envKey, envValue)
 }
 
+// RevokableSetenvs ...
+func RevokableSetenvs(envs map[string]string) (func() error, error) {
```

Comment:
NOTE: with a slight modification you can solve to store and restore only the envs, which was set without error (it also takes less iteration over envs):

```
	origValues := map[string]string{}

	revokeFn := func() error {
		for k, v := range origValues {
			if err := os.Setenv(k, v); err != nil {
				return err
			}
		}
		return nil
	}

	for k, v := range envs {
		origValue := os.Getenv(k)

		if err := os.Setenv(k, v); err != nil {
			return revokeFn, err
		}

		origValues[k] = origValue
	}

	return revokeFn, nil
```